device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -820,21 +901,80 @@ private void handleDisconnection(TransportException transportException)
         reconnect(transportException);
     }
 
+    // should only be called when multiplexing an only a particular device went offline
+    private void reconnectDeviceSession(TransportException transportException, String deviceId)
+    {
+        long reconnectionStartTimeMillis = System.currentTimeMillis();
+        int reconnectionAttempt = 0;
+        boolean hasReconnectOperationTimedOut = this.hasOperationTimedOut(reconnectionStartTimeMillis, deviceId);
+        RetryDecision retryDecision = null;
+
+        while (this.deviceConnectionStates.get(deviceId) == IotHubConnectionStatus.DISCONNECTED_RETRYING
+                && !hasReconnectOperationTimedOut)
+        {
+            this.log.trace("Attempting device session reconnect attempt {}", reconnectionAttempt);
```

Comment:
Poor grammar. Maybe this?
"Attempting device session reconnect: attempt {}"

Reply:
How about "Attempting to reconnect device session: attempt {}"?

Reply:
sure